src/back/wgsl/writer.rs

```diff
@@ -1203,10 +1356,11 @@ impl<W: Write> Writer<W> {
     /// Ends in a newline
     fn write_global_constant(
         &mut self,
-        constant: &Constant,
+        module: &Module,
+        inner: &ConstantInner,
         handle: Handle<Constant>,
     ) -> BackendResult {
-        match constant.inner {
+        match *inner {
```

Comment:
We generally write scalar constants before the types, and composite constants after the types. This is needed because some array types may need to have the length of scalar constants, but the composite constants always need types to be defined.

Reply:
Here we write only global constants and writes types for scalar constants and composite.